packages/ui/src/components/TagInput/index.tsx

```diff
@@ -17,79 +37,82 @@ const STATUS = {
 type Keys = keyof typeof STATUS
 type StatusValue = (typeof STATUS)[Keys]
 
-const variants = {
-  base: ({ theme: { colors, shadows, radii } }: { theme: Theme }) => `
-    padding: 8px;
-    cursor: text;
-    border-radius: ${radii.default};
-    border: 1px solid ${colors.neutral.border};
-    &:focus-within {
-      border: 1px solid ${colors.primary.border};
-      box-shadow: ${shadows.focusPrimary};
-    }
+type TagInputContainersProps = {
+  size: TagInputSize
+}
+const TagInputContainer = styled('div', {
+  shouldForwardProp: prop => !['size'].includes(prop),
+})<TagInputContainersProps>`
+  display: flex;
+  gap: ${({ theme }) => theme.space['1']};
+  background-color: ${({ theme: { colors } }) => colors.neutral.background};
+  min-height: ${({ size }) => TAGINPUT_SIZE_HEIGHT[size]}px;
 
-    &:hover {
-      border: 1px solid ${colors.primary.border};
-    }
+  padding: ${({ theme, size }) =>
+    `${theme.space[TAGINPUT_SIZE_PADDING[size]]} ${theme.space['2']}`};
+  cursor: text;
 
-    & > * {
-      margin: 6px;
-    }
-  `,
-  bordered: ({ theme: { shadows } }: { theme: Theme }) => `
-    margin-top: 0;
-    padding: 8px 0;
+  background: ${({ theme }) => theme.colors.neutral.background};
+  border: 1px solid ${({ theme }) => theme.colors.neutral.border};
+  border-radius: ${({ theme }) => theme.radii.default};
 
-    > input:focus {
-      box-shadow: ${shadows.focusPrimary};
-    }
+  &:focus-within {
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+    box-shadow: ${({ theme }) => theme.shadows.focusPrimary};
+  }
 
-    > * {
-      margin-bottom: 6px;
-      &:not(:last-child) {
-        margin-right: 6px;
-      }
-    }
-  `,
-  'no-border': ({ theme: { shadows } }: { theme: Theme }) => `
-    &:focus-within {
-      box-shadow: ${shadows.focusPrimary};
-    }
+  &[data-success='true'] {
+    border-color: ${({ theme }) => theme.colors.success.border};
+  }
 
-    > * {
-      margin-right: 6px;
-      margin-bottom: 6px;
-    }
-  `,
-} as const
+  &[data-error='true'] {
+    border-color: ${({ theme }) => theme.colors.danger.border};
+  }
 
-type Variant = keyof typeof variants
+  &:hover {
+    border-color: ${({ theme }) => theme.colors.primary.borderHover};
+  }
 
-type TagInputContainersProps = {
-  variant: Variant
-}
+  &[data-readonly='true'] {
+    border-color: ${({ theme }) => theme.colors.neutral.border};
+    background: ${({ theme }) => theme.colors.neutral.backgroundWeak};
+    cursor: not-allowed;
```

Comment:
I think we should allow cursor in ready only as you should be able to select and copy the content